impl/maven-impl/src/main/java/org/apache/maven/impl/model/DefaultModelProcessor.java

```diff
@@ -77,17 +79,19 @@ public DefaultModelProcessor(ModelXmlFactory modelXmlFactory, @Nullable List<Mod
         this.modelParsers = modelParsers;
     }
 
+    /**
+     * @implNote
+     * The ModelProcessor#locatePom never returns null while the ModelParser#locatePom needs to return an existing path!
+     */
     @Override
     public Path locateExistingPom(Path projectDirectory) {
-        // Note that the ModelProcessor#locatePom never returns null
-        // while the ModelParser#locatePom needs to return an existing path!
         Path pom = modelParsers.stream()
                 .map(m -> m.locate(projectDirectory)
                         .map(org.apache.maven.api.services.Source::getPath)
                         .orElse(null))
                 .filter(Objects::nonNull)
                 .findFirst()
-                .orElseGet(() -> doLocateExistingPom(projectDirectory));
```

Comment:
we have already very nice functional clean code style. very good codebase. func over imper

Reply:
You changed the semantics, it's wrong.